sub/mutation_events.go

```diff
@@ -15,17 +15,22 @@ import (
 type ProtoMutationEventHandlerFn[T proto.Message] func(context.Context, models.ProtoMutationEvent[T]) error
 
 // NewMutationEventSqsProcessor creates an SQS event processor that reads mutation events
-// from an SQS queue, Unmarshal them into protocol buffer messages, and processes them
-// using a strongly-typed event handler function.
+// from an SQS queue, unmarshal them into strongly typed protobuf messages, and processes them
+// using the provided event handler. If includeSnsWrapper is true, messages are unwrapped from
+// SNS-compatible envelopes before processing.
 func NewMutationEventSqsProcessor[T proto.Message](
 	svc *sqs.Client,
 	queueURL string,
 	newMessage func() T,
 	handler ProtoMutationEventHandlerFn[T],
+	includeSnsWrapper bool,
 ) *SqsEventProcessor {
+	var snsWrapper func(context.Context, SnsWrapper) error
 
-	snsString := MutationEventHandlerToStringHandler(handler, newMessage)
-	snsWrapper := StringHandlerToSnsWrapperHandler(snsString)
+	snsEventMutationMsg := MutationEventHandlerToStringHandler(handler, newMessage)
```

Comment:
This name seems a bit confusing.
```suggestion
	sqsEventMutationMsg := MutationEventHandlerToStringHandler(handler, newMessage)
```